### DISCUSSION TITLE
[mLRS] Uploading Mission to INAV 8

### DISCUSSION THREAD

**Discussion by MiguelitoBeard** [stronnag/mwptools#183] [General]
Hi, I am using mLRS (Matek H/W) and have INAV 8 working. MWP tools connects over BT and I see telemetry. However when I create a simple mission, all the controls to load a mission to the FC are greyed out. What am I doing wrong? Thanks!!!

**Comment by stronnag** [comment]
What you're doing wrong is failing to provide any useful information.

https://www.daria.co.uk/mwptools/mwp_support/

https://www.daria.co.uk/mwptools/mwp_support/#information-requirements

**Reply by MiguelitoBeard** [reply]
OK, sorry :) Attached is the log. 
[mwp_stderr_2024-12-02.txt](https://github.com/user-attachments/files/17983298/mwp_stderr_2024-12-02.txt)
I can successfully connect via BT to the Matek eLRS module that is in turn is connected to INAV B running on a Matek FC. When this connection is made all the telemetry seems to appear in MWP and I can see see telemetry information change.  Then I create a simple mission in the user interface. I can save that mission but when I try to upload the mission to the FC the option in the UI is greyed out and not available.  I have also attached a screenshot so you can see. 
![mwp](https://github.com/user-attachments/assets/7dd7d227-8865-41dd-a2a6-babc08d06ab0)
Thanks!

**Reply by b14ckyy** [reply]
See my comment below. Thats a issue caused by mLRS "misbehaving" in terms of MSP specifications. This is fixed in latest code but you need to compile yourself or ask me to compile a firmware update for you :)

**Comment by b14ckyy** [comment]
you need to compile mLRS firmware from latest MAIN branch. The provided pre build beta versions miss some cruicial changes to work with MWP. If you tell me EXACTLY what mLRS hardware you use as Tx and Rx I can compile you the firmware

**Reply by MiguelitoBeard** [reply]
Wow thanks! I am currently using 1.3.03 on the mLRS MAVLink 900MHz TX Module KIT, mR900-30-TX and mLRS MAVLink 900MHz Receiver, mR900-30

**Reply by b14ckyy** [reply]
Here you go. Tx and Rx firmware. FLash with full erase and set the settings from scratch to avoid any conflicts. Then Rebind with 4s button press (even if it autobinds, rebind again). 
[matek-mr900-30-g431kb.zip](https://github.com/user-attachments/files/17984151/matek-mr900-30-g431kb.zip)

**Reply by MiguelitoBeard** [reply]
Will try tomorrow AM. Thanks!

**Reply by MiguelitoBeard** [reply]
Just tried and worked! Thanks! I was able to create and save a mission to EPROM in MWP and loaded back into INAV 8. Now will try MWP in the field, looking great and thanks to all for the Development and help :)

**Comment by stronnag** [comment]
@MiguelitoBeard Thanks for the log, it confirms the known non-compliant behaviour of older mLRS firmware.
@b14ckyy Thanks for sorting this out. I suspect we will see more of this kind of issue. 

Added mLRS to the title in the (probably misguided) hope that people will come here first rather than raising new issues.

**Reply by b14ckyy** [reply]
Got answer from OlliW that he is planning on a 1.4.04 beta release soon. When this is out, this won't be a problem anymore. For now I guess I will just prepare a pack of firmwares to have it available 😁
